studybuddies/src/app/components/tutor/create-class-tutor/create-class-tutor.component.ts

```diff
@@ -7,9 +11,67 @@ import { Component, OnInit } from '@angular/core';
 })
 export class CreateClassTutorComponent implements OnInit {
 
-  constructor() { }
+  constructor(public roomService: RoomService, private router: Router) { }
 
   ngOnInit(): void {
   }
 
+  validate(form: NgForm){
+
+    let iDay = form.value.iTime.split(":");
+    let fDay = form.value.fTime.split(":");
+    let date = form.value.date.split("-");
+    let isCorrect;
+    let now = new Date();
+
+    if (parseInt(iDay[0]) > parseInt(fDay[0]) || parseInt(iDay[0])==parseInt(fDay[0]) && parseInt(iDay[1]) >= parseInt(fDay[1])) {
+      isCorrect = false;
+    } else if(parseInt(date[0]) < now.getFullYear() || parseInt(date[0]) == now.getFullYear() && parseInt(date[1]) < now.getMonth()+1 || parseInt(date[0]) == now.getFullYear() && parseInt(date[1]) == now.getMonth()+1 && parseInt(date[2]) < now.getDate()){
+      isCorrect = false;
+    }else {
+      isCorrect = true;
+    }
+
+    return isCorrect;
+  }
+
+  createRoom(form: NgForm){
+
+    console.log(form.value.date);
+    console.log(form.value.iTime);
+
+    let date = form.value.date.split("-");
+    let iDay = form.value.iTime.split(":");
+    let fDay = form.value.fTime.split(":");
+
+    let room: Class = {
+      id_user : "",
+      description : form.value.description,
+      university : form.value.university,
+      degree : form.value.degree,
+      subject : form.value.subject,
+      starting_time : new Date(parseInt(date[0]), parseInt(date[1])-1, parseInt(date[2]),
+      parseInt(iDay[0]), parseInt(iDay[1])),
+      ending_time : new Date(parseInt(date[0]), parseInt(date[1])-1, parseInt(date[2]),
+      parseInt(fDay[0]), parseInt(fDay[1])),
+      price_per_hour : form.value.money,
+      is_private : true,
+      date : new Date(),
+      iTime : new Date(),
+      fTime : new Date()
+    };
+    if(this.validate(form)){
+      this.roomService.createRoom(room).subscribe(
+        res => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Frontend/pullRequest?prid=7203289)
```suggestion
        (res) => {
```